uaclient/actions.py

```diff
@@ -218,3 +219,52 @@ def collect_logs(cfg: config.UAConfig, output_dir: str):
             system.write_file(
                 os.path.join(output_dir, os.path.basename(f)), content
             )
+
+
+def should_disable_auto_attach(cfg: config.UAConfig) -> bool:
+    disable_auto_attach = util.is_config_value_true(
+        config=cfg.cfg, path_to_value="features.disable_auto_attach"
+    )
+    if disable_auto_attach:
+        msg = "Skipping auto-attach. Config disable_auto_attach is set."
+        logging.debug(msg)
+        print(msg)
+
+    return disable_auto_attach
+
+
+def get_cloud_instance(
+    cfg: config.UAConfig,
+) -> Optional[AutoAttachCloudInstance]:
```

Comment:
We raise an error in this function if the instance is None
```suggestion
) -> AutoAttachCloudInstance:
```